csharp/autobuilder/Semmle.Autobuild.CSharp/StandaloneBuildRule.cs

```diff
@@ -9,44 +9,34 @@ namespace Semmle.Autobuild.CSharp
     internal class StandaloneBuildRule : IBuildRule<CSharpAutobuildOptions>
     {
         private readonly string? dotNetPath;
+        private readonly IDictionary<string, string>? env;
 
-        internal StandaloneBuildRule(string? dotNetPath)
+        internal StandaloneBuildRule(string? dotNetPath, IDictionary<string, string>? env)
         {
             this.dotNetPath = dotNetPath;
+            this.env = env;
         }
 
         public BuildScript Analyse(IAutobuilder<CSharpAutobuildOptions> builder, bool auto)
         {
-            BuildScript GetCommand()
+            if (!builder.Options.Buildless
+                || builder.CodeQLExtractorLangRoot is null
+                || builder.CodeQlPlatform is null)
             {
-                string standalone;
-                if (builder.CodeQLExtractorLangRoot is not null && builder.CodeQlPlatform is not null)
-                {
-                    standalone = builder.Actions.PathCombine(builder.CodeQLExtractorLangRoot, "tools", builder.CodeQlPlatform, "Semmle.Extraction.CSharp.Standalone");
-                }
-                else
-                {
-                    return BuildScript.Failure;
-                }
-
-                var cmd = new CommandBuilder(builder.Actions);
-                cmd.RunCommand(standalone);
-
-                if (!string.IsNullOrEmpty(this.dotNetPath))
-                {
-                    cmd.Argument("--dotnet");
-                    cmd.QuoteArgument(this.dotNetPath);
-                }
-
-                return cmd.Script;
+                return BuildScript.Failure;
             }
 
-            if (!builder.Options.Buildless)
+            var standalone = builder.Actions.PathCombine(builder.CodeQLExtractorLangRoot, "tools", builder.CodeQlPlatform, "Semmle.Extraction.CSharp.Standalone");
+            var cmd = new CommandBuilder(builder.Actions, environment: this.env);
```

Comment:
`environment: this.env` is new in this line. I think the environment variables assembled in the autobuilder should be passed towards to the the standalone extractor. (At least this is what we do in case of autobuild)

This is related to https://github.com/github/codeql/pull/15509.